public/sw.js

```diff
@@ -56,17 +56,23 @@ const fetchFromNetwork = (event) => {
 			caches.open(version).then((cache) => cache.put(event.request, clonedResponse));
 		}
 		return response;
-	}).catch(() => {
-		if (hasHash(event.request.url)) { return caches.match(event.request.url); }
-		// If the request URL hasn't been served from cache and isn't sockjs we suppose it's HTML
-		if (!/\/sockjs\//.test(event.request.url)) { return caches.match(HTMLToCache); }
-		// Only for sockjs
-		return new Response('No connection to the server', {
-			status: 503,
-			statusText: 'No connection to the server',
-			headers: new Headers({ 'Content-Type': 'text/plain' }),
-		});
-	});
+	}).catch(() =>
+		new Promise(function(resolve) {
+			caches.match(event.request.url).then((cached) => {
```

Comment:
@Shailesh351 We are already checking cache in previous function, maybe use that to avoid another call.

Reply:
Thanks @shubhsherl 🙌  This is actually better way 🥂